core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3474,14 +3474,7 @@ private void scheduleAsyncPersistenceInternal(LockedInodePath inodePath,
     }
   }
 
-  /**
-   * Actively sync metadata, based on a list of changed files.
-   *
-   * @param path the path to sync
-   * @param changedFiles collection of files that are changed under the path to sync, if this is
-   *        null, force sync the entire directory
-   * @param executorService executor to execute the parallel incremental sync
-   */
+  @Override
   public void activeSyncMetadata(AlluxioURI path, Collection<AlluxioURI> changedFiles,
       ExecutorService executorService) throws IOException {
     if (changedFiles == null) {
```

Comment:
Please preserve the comment on the behavior when `changedFiles` is `null`. Also please add a `@Nullable` annotation on this parameter.

Reply:
Actually this is more complicated: when `changedFiles` is not `null` but an empty collection, this method does nothing. It starts a full sync when `changedFiles` is `null`. Should update the comments to document this behavior.